### PR TITLE
Remove unnecessary gtest for performance tests

### PR DESCRIPTION
**Pull request by streeve**
Benchmarks do not use gtest

### CODE REVIEW

**Review by dalg24 — APPROVED** [review]

File: benchmark/cajita/CMakeLists.txt [hunk 1]
```diff
@@ -22,14 +22,11 @@ if(Kokkos_ENABLE_SERIAL OR Kokkos_ENABLE_CUDA OR Kokkos_ENABLE_OPENMP OR Kokkos_
     endif()
 
   if(Cabana_ENABLE_TESTING)
```
Comment (dalg24) [inline benchmark/cajita/CMakeLists.txt:24]: By the way why are we guarding with ` if(Cabana_ENABLE_TESTING)`?

**Review by junghans — APPROVED** [review]

File: benchmark/cajita/CMakeLists.txt (hunk 1 above)
Comment (junghans) [reply benchmark/cajita/CMakeLists.txt:24]: You mean we should use `BUILD_TESTING` instead?

Comment (dalg24) [reply benchmark/cajita/CMakeLists.txt:24]: I wouldn't put any condition (so yes rely on `BUILD_TESTING`)

Comment (streeve) [reply benchmark/cajita/CMakeLists.txt:24]: I usually either want to build the benchmark only or build tests for CI. And we only enable tests with `Cabana_ENABLE_TESTING` on

**Pull request merged**